security/src/main/java/io/micronaut/security/handlers/RedirectRejectionHandler.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * @author Sergio del Amo
+ * @since 1.0
+ */
+
+package io.micronaut.security.handlers;
+
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.async.publisher.Publishers;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.MutableHttpResponse;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * If beans {@link UnauthorizedRejectionUriProvider} and {@link ForbiddenRejectionUriProvider} exists provides
+ * a {@link RejectionHandler} implementation which performs redirects.
+ *
+ * It can be used for session based authentication flows or Open ID flows.
+ *
+ * @author Sergio del Amo
+ * @since 1.0
+ */
+@Singleton
+@Replaces(HttpStatusCodeRejectionHandler.class)
+@Requires(beans = {UnauthorizedRejectionUriProvider.class, ForbiddenRejectionUriProvider.class})
+public class RedirectRejectionHandler implements RejectionHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(RedirectRejectionHandler.class);
+
+    private final UnauthorizedRejectionUriProvider unauthorizedRejectionUriProvider;
+    private final ForbiddenRejectionUriProvider forbiddenRejectionUriProvider;
+
+    /**
+     *
+     * @param unauthorizedRejectionUriProvider URI Provider to redirect to if unauthenticated
+     * @param forbiddenRejectionUriProvider URI Provider to redirect to if authenticated but not enough authorization level.
+     */
+    public RedirectRejectionHandler(UnauthorizedRejectionUriProvider unauthorizedRejectionUriProvider,
```

Comment:
Please add javadoc descriptions for all public and protected methods and constructors